src/vs/workbench/contrib/interactive/browser/interactiveEditor.ts

```diff
@@ -154,6 +148,12 @@ export class InteractiveEditor extends EditorPane {
 
 		codeEditorService.registerDecorationType('interactive-decoration', DECORATION_KEY, {});
 		this._register(this.#keybindingService.onDidUpdateKeybindings(this.#updateInputDecoration, this));
+		this._register(this.#notebookExecutionStateService.onDidChangeCellExecution((e) => {
+			const cell = this.#notebookWidget.value?.getCellByHandle(e.cellHandle);
```

Comment:
This is where getCellByHandle is necessary. The cellExecution change event only has the cell handle for finding the cell.